catalog/cataloger_diff.go

```diff
@@ -168,11 +174,26 @@ func (c *diffEffectiveCommits) ParentEffectiveCommitByBranchID(branchID int64) C
 	return c.ParentEffectiveCommit
 }
 
-func newDiffResultsBatchWriter(tx db.Tx, tableName string) *diffResultsBatchWriter {
+func newDiffResultsBatchWriter(tx db.Tx, tableName string, additionalFields []string) *diffResultsBatchWriter {
+	// scan Entry db metadata and map addinalFields names into the struct field index
+	fieldsByIndex := make(map[int]string)
+	if len(additionalFields) > 0 {
+		ft := reflect.TypeOf(Entry{})
+		for _, fieldName := range additionalFields {
+			for i, l := 0, ft.NumField(); i < l; i++ {
+				// lookup db tag value on field - when matched we keep the field position at the same index of the field name
+				if ft.Field(i).Tag.Get("db") == fieldName {
+					fieldsByIndex[i] = fieldName
```

Comment:
This fails silently when a field name is not found,  I would much rather fail noisily, even to the extent of killing the SQL query with a nasty SQL error.  At the very least, if nothing is found, log an error here.